http/src/main/scala/org/http4s/blaze/http/http2/Http2Result.scala

```diff
@@ -0,0 +1,23 @@
+package org.http4s.blaze.http.http2
+
+/** Result type of many of the codec methods */
+sealed trait Http2Result
```

Comment:
These are probably not ever in a collection, but extending `Product with Serializable` in the trait improves inference.  I've tried to get into that habit everywhere.